src/main/java/com/example/report/service/ReportService.java

```diff
@@ -119,11 +133,120 @@ public List<ReportEntryDTO> generateReportEntries(LocalDate from, LocalDate to)
+        headerStyle.setFont(font);
+        headerStyle.setAlignment(HorizontalAlignment.CENTER);
+        headerStyle.setVerticalAlignment(VerticalAlignment.CENTER);
+        headerStyle.setBorderBottom(BorderStyle.THIN);
+        headerStyle.setBottomBorderColor(HSSFColor.HSSFColorPredefined.TEAL.getIndex());
+        headerStyle.setWrapText(true);
+
+        Row header = sheet.createRow(0);
+        Cell headerCell = header.createCell(0);
+        headerCell.setCellValue("Report ID");
+        headerCell.setCellStyle(headerStyle);
+
+        headerCell = header.createCell(1);
+        headerCell.setCellValue("Start date");
+        headerCell.setCellStyle(headerStyle);
+
+        headerCell = header.createCell(2);
+        headerCell.setCellValue("End date");
+        headerCell.setCellStyle(headerStyle);
+        return headerStyle;
+    }
+
+    private CellStyle cellNumberStyle(Workbook workbook){
+        CellStyle numberStyle = workbook.createCellStyle();
+        numberStyle.setWrapText(true);
+        numberStyle.setBorderBottom(BorderStyle.THIN);
+        numberStyle.setDataFormat((short) 1);
+        numberStyle.setAlignment(HorizontalAlignment.LEFT);
+        numberStyle.setBottomBorderColor(HSSFColor.HSSFColorPredefined.TEAL.getIndex());
+        return numberStyle;
+    }
+
+    private CellStyle cellDateStyle(Workbook workbook){
+        CellStyle dateStyle = workbook.createCellStyle();
+        dateStyle.setWrapText(true);
+        dateStyle.setBorderBottom(BorderStyle.THIN);
+        dateStyle.setDataFormat((short) 14);
+        dateStyle.setBottomBorderColor(HSSFColor.HSSFColorPredefined.TEAL.getIndex());
+        return dateStyle;
+    }
+
+    public ResponseEntity<ByteArrayResource> extractAllReports() throws IOException {
+        List<Report> reportList = reportRepository.findAll();
+
+        Workbook workbook = new XSSFWorkbook();
+        Sheet sheet = workbook.createSheet("List of reports");
+        applyCommonSheetStyle(sheet);
+
+        headerCellStyle(workbook, sheet);
+        CellStyle numberStyle = cellNumberStyle(workbook);
+        CellStyle dateStyle = cellDateStyle(workbook);
+
+        for (int i = 0; i < reportList.size(); i++) {
+            Report report = reportList.get(i);
+            writeReportDataInExcelRow(sheet,i+1,report, numberStyle, dateStyle);
+        }
+
+        DateTimeFormatter myFormatObj = DateTimeFormatter.ofPattern("YYYYMMdd_HHmm");
+        String localDateTime = LocalDateTime.now().format(myFormatObj).toString();
+
+        ByteArrayOutputStream stream = new ByteArrayOutputStream();
+        HttpHeaders header = new HttpHeaders();
+        header.setContentType(new MediaType("application", "force-download"));
+        header.set(HttpHeaders.CONTENT_DISPOSITION, String.format("attachment; filename=List_of_Reports_%s.xlsx", localDateTime));
+        workbook.write(stream);
+        workbook.close();
+
+        return new ResponseEntity<>(new ByteArrayResource(stream.toByteArray()),
```

Comment:
Let's refactor this method to return the workbook and we can take care of creating entity and setting headers in the controller